### PR TITLE
Issue #7858 - GzipHandler request.isHandled support

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java [hunk 1]
```diff
@@ -616,6 +616,9 @@ public void setInflateBufferSize(int size)
     @Override
     public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
     {
+        if (response.isCommitted() || baseRequest.isHandled())
```
Comment (gregw) [inline jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java]: I know we do sometimes check isCommitted elsewhere, but I think isHandled is enough here.

**Review by gregw — CHANGES_REQUESTED** [review]

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java (hunk 1 above)
Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java]: Actually, this is a bad if statement overall.

Have to rethink this, as we don't un-gzip request content with this if statement here.

Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java]: but if the request is handled, do we even want to?

Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java]: I submitted an alternative approach.

**Review by gregw — DISMISSED** [review]

**Comment by joakime** [comment]
The tests on CI failed, as the expectations were incorrect.
I improved the expectations and documented the reasoning.
This will need to be re-reviewed

**Review by gregw — APPROVED** [review]

**Pull request merged**